src/bin/cargo/commands/describe_future_incompatibilities.rs

```diff
@@ -0,0 +1,43 @@
+use crate::command_prelude::*;
+use cargo::drop_println;
+use std::io::BufRead;
+use serde::Deserialize;
+use cargo::core::compiler::future_incompat::{FutureIncompatReportLine, FUTURE_INCOMPAT_FILE};
+use anyhow::anyhow;
+
+pub fn cli() -> App {
+    subcommand("describe-future-incompatibilities")
+        .about("Reports any crates which will eventually stop compiling")
+}
+
+
+pub fn exec(config: &mut Config, args: &ArgMatches<'_>) -> CliResult {
+    if !config.cli_unstable().future_incompat_report {
+        return Err(anyhow!("`cargo describe-future-incompatibilities` can only be used on the nightly channel \
+            with `-Z future-incompat-report`").into());
+    }
+    let ws = args.workspace(config)?;
+    let report = ws.target_dir().open_ro(FUTURE_INCOMPAT_FILE, ws.config(), "Future incompatible report")?;
+    let mut lines = Vec::new();
+    let mut has_end = false;
+    for line in std::io::BufReader::new(report.file()).lines() {
+        let line: FutureIncompatReportLine = serde_json::from_str(line.unwrap().as_str()).unwrap();
+        match line {
+            FutureIncompatReportLine::CrateReport(report) => lines.push(report),
+            FutureIncompatReportLine::Finished => has_end = true,
+        };
+    }
+    lines.sort_by(|r1, r2| r1.name.cmp(&r2.name));
```

Comment:
Nit: you can do `sort_by_key` her

Reply:
That would require cloning the `name` value

Reply:
Good point, case rested.